error: expected identifier, found keyword `self`
 --> $DIR/ref-self.rs:8:23
  |
8 |     fn method_ref(ref self: Pin<&mut Self>) {} //~ ERROR expected identifier, found keyword `self`
  |                       ^^^^ expected identifier, found keyword

error: expected identifier, found keyword `self`
 --> $DIR/ref-self.rs:9:31
  |
9 |     fn method_ref_mut(ref mut self: Pin<&mut Self>) {} //~ ERROR expected identifier, found keyword `self`
  |                               ^^^^ expected identifier, found keyword
