error: Trait bound inner::Inner: std::marker::Unpin does not depend on any type or lifetime parameters
  --> $DIR/trivial_bounds.rs:15:37
   |
15 |     impl Unpin for Foo where Inner: Unpin {} //~ ERROR std::marker::Unpin does not depend on any type or lifetime parameters
   |                                     ^^^^^
   |
note: lint level defined here
  --> $DIR/trivial_bounds.rs:6:9
   |
6  | #![deny(trivial_bounds)]
   |         ^^^^^^^^^^^^^^

error: Trait bound inner::Wrapper<inner::Inner>: std::marker::Unpin does not depend on any type or lifetime parameters
  --> $DIR/trivial_bounds.rs:23:46
   |
23 |     impl Unpin for Bar where Wrapper<Inner>: Unpin {} //~ ERROR std::marker::Unpin does not depend on any type or lifetime parameters
   |                                              ^^^^^
