error[E0308]: mismatched types
  --> $DIR/cfg_attr-type-mismatch.rs:19:27
   |
19 |     let _: Pin<&mut u8> = x.inner; //~ ERROR E0308
   |                           ^^^^^^^ expected struct `std::pin::Pin`, found &mut u8
   |
   = note: expected type `std::pin::Pin<&mut u8>`
              found type `&mut u8`

error[E0308]: mismatched types
  --> $DIR/cfg_attr-type-mismatch.rs:23:22
   |
23 |     let _: &mut u8 = x.inner; //~ ERROR E0308
   |                      ^^^^^^^
   |                      |
   |                      expected &mut u8, found struct `std::pin::Pin`
   |                      help: consider mutably borrowing here: `&mut x.inner`
   |
   = note: expected type `&mut u8`
              found type `std::pin::Pin<&mut u8>`
