Index: parking-lot-core/Cargo.toml
===================================================================
--- parking-lot-core.orig/Cargo.toml
+++ parking-lot-core/Cargo.toml
@@ -44,12 +44,5 @@ optional = true
 [features]
 deadlock_detection = ["petgraph", "thread-id", "backtrace"]
 nightly = []
-[target."cfg(target_os = \"cloudabi\")".dependencies.cloudabi]
-version = "0.1.0"
-[target."cfg(target_os = \"redox\")".dependencies.redox_syscall]
-version = "0.1.56"
 [target."cfg(unix)".dependencies.libc]
 version = "0.2.71"
-[target."cfg(windows)".dependencies.winapi]
-version = "0.3.8"
-features = ["winnt", "ntstatus", "minwindef", "winerror", "winbase", "errhandlingapi", "handleapi"]
