error[E0119]: conflicting implementations of trait `FooMustNotImplDrop` for type `Foo<_, _>`:
 --> $DIR/conflict-drop.rs:4:1
  |
4 | #[pin_project] //~ ERROR E0119
  | ^^^^^^^^^^^^^^
  | |
  | first implementation here
  | conflicting implementation for `Foo<_, _>`

error[E0119]: conflicting implementations of trait `std::ops::Drop` for type `Bar<_, _>`:
  --> $DIR/conflict-drop.rs:15:1
   |
15 | #[pin_project(PinnedDrop)] //~ ERROR E0119
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `Bar<_, _>`
...
27 | impl<T, U> Drop for Bar<T, U> {
   | ----------------------------- first implementation here
