error[E0308]: mismatched types
  --> $DIR/remove-attr-from-field.rs:27:38
   |
27 |     let _: Pin<&mut PhantomPinned> = x.field; //~ ERROR E0308
   |                                      ^^^^^^^ expected struct `std::pin::Pin`, found mutable reference
   |
   = note: expected type `std::pin::Pin<&mut std::marker::PhantomPinned>`
              found type `&mut std::marker::PhantomPinned`

error[E0308]: mismatched types
  --> $DIR/remove-attr-from-field.rs:31:38
   |
31 |     let _: Pin<&mut PhantomPinned> = x.field; //~ ERROR E0308
   |                                      ^^^^^^^ expected struct `std::pin::Pin`, found mutable reference
   |
   = note: expected type `std::pin::Pin<&mut std::marker::PhantomPinned>`
              found type `&mut std::marker::PhantomPinned`
